base/lthooks.dtx

```diff
@@ -3116,27 +3146,13 @@
           {
 %    \end{macrocode}
 %    If the label is |top-level| then clear the token list, as all code
-%    there is under the same label.  Marked removal is not implemented
-%    for |top-level| because it is hard to reliably know that no code
-%    was added to \cs{@@_toplevel\textvisiblespace\meta{hook}} (granted that
-%    an empty code could be interpreted as that, but then it differs in
-%    behaviour from other labels, in which an empty chunk is still valid
-%    for removal).  Besides, it doesn't make much (if any) sense for
-%    packages to remove |top-level| code.  So here the chunk is just
-%    cleared unconditionally.
+%    there is under the same label.
 %    \begin{macrocode}
             \str_if_eq:nnTF {#2} { top-level }
               { \@@_tl_gclear:c { @@_toplevel~#1 } }
               {
-%    \end{macrocode}
-%    Otherwise check if the label being removed exists in the code pool.
-%    If it does, just call \cs{@@_gremove_code_do:nn} to do the removal,
-%    otherwise mark it to be removed.
-%    \begin{macrocode}
-                \prop_get:cnNTF { g_@@_#1_code_prop } {#2} \l_@@_return_tl
-                  { \@@_gremove_code_do:nn }
-                  { \@@_mark_removal:nn }
-                      {#1} {#2}
+                \prop_gpop:cnNF { g_@@_#1_code_prop } {#2} \l_@@_return_tl
+                  { \msg_error:nnnn { hooks } { cannot-remove } {#1} {#2} }
```

Comment:
strange space before `\l_@@retun_tl` (maybe only in the diff)

Reply:
I only see the space between `{#2}` and `\l_@@_return_tl`, but I suppose that's not what you mean?

Reply:
I was only showing strangely in the view I used I guess